src/modules/auth/auth.service.ts

```diff
@@ -81,4 +89,61 @@ export class AuthService {
       userId: user.id,
     };
   }
+
+  async forgotPassword({ email }: ForgotPasswordDto) {
+    const user = await this.usersRepository.findOneBy({ email });
+    if (!user) {
+      throw new HttpException(
+        {
+          status: HttpStatus.BAD_REQUEST,
+          error: 'Email address is not valid',
+        },
+        HttpStatus.BAD_REQUEST,
+      );
+    } else {
+      const hash = randomBytes(20).toString('hex'); //create random hashed link
+
+      await this.forgotPasswordRepository.save({
+        user: user,
+        link: hash,
+      });
+
+      const url = process.env.RESET_PASSWORD_URL + '?token=' + hash;
+
+      return await this.mailerService.sendMail({
+        to: email,
+        subject: 'Devs Heads restore password',
+        from: 'surkovdavid@gmail.com',
+        html: `<h1>Change password</h1><p>If you want to reset your password click:</p><a href="${url}">${url}</a>`,
+      });
+    }
+  }
+  async restorePassword({ token, password }: RestorePasswordDto) {
+    const { user } = await this.forgotPasswordRepository.findOne({
+      where: {
+        link: token,
+      },
+      relations: {
+        user: true,
+      },
+    });
+
+    if (user) {
+      const newPassword = bcrypt.hashSync(password, 8);
```

Comment:
no magic numbers, move to constant and gave obvious name

Reply:
done